From f91b21db2c57127bdc92336729ebf84f6d6a4e8a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?= <avarab@gmail.com>
Date: Sat, 17 Jul 2010 15:48:07 +0000
Subject: [PATCH] git submodule add: Remove old docs about implicit -f
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

git submodule add no longer implicitly adds with --force. Remove
references to the old functionality in the documentation.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---

> To make the behavior of "git submodule add" more consistent with "git add"
> ignored submodule paths should not be silently added when they match an
> entry in a .gitignore file. To be able to override that default behavior
> in the same way as we can do that for "git add", the new option "--force"
> is introduced.
>
> Signed-off-by: Jens Lehmann <Jens.Lehmann@web.de>

You beat me to it. I like this better than the original functionality
that I implemented, and it's now possible with --ignore-missing.

So,:

    Acked-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>

This amended patch removes some obsolete documentation that you
missed.

> +               echo >&2 "The following path is ignored by one of your .gitignore files:" &&
> +               echo >&2 $path &&
> +               echo >&2 "Use -f if you really want to add it."

Is the "it" intentional? We currently say "them" in git add regardless
of how many things are being added, so perhaps we should say "it"
there too for singulars.

I'll jot this down as something to look at for plural support once
gettext gets merged.

 Documentation/git-submodule.txt |    4 ----
 1 files changed, 0 insertions(+), 4 deletions(-)

diff --git a/Documentation/git-submodule.txt b/Documentation/git-submodule.txt
index 617069f..1ed331c 100644
--- a/Documentation/git-submodule.txt
+++ b/Documentation/git-submodule.txt
@@ -95,10 +95,6 @@ is the superproject and submodule repositories will be kept
 together in the same relative location, and only the
 superproject's URL needs to be provided: git-submodule will correctly
 locate the submodule using the relative URL in .gitmodules.
-+
-The submodule will be added with "git add --force <path>". I.e. git
-doesn't care if the new path is in a `gitignore`. Your invocation of
-"git submodule add" is considered enough to override it.
 
 status::
 	Show the status of the submodules. This will print the SHA-1 of the
-- 
1.7.2.rc3.125.g94e09.dirty

